client/state/login/oauth2/reducer.js

```diff
@@ -115,6 +115,7 @@ export const clients = createReducer( initialClientsData, {
 
 export const currentClientId = createReducer( null, {
 	[ ROUTE_SET ]: ( state, { path, query } ) => startsWith( path, '/log-in' ) && query.client_id || null,
+	[ ROUTE_SET ]: ( state, { path, query } ) => startsWith( path, '/start' ) && query.client_id || null,
```

Comment:
Should we move this to a new branch of the state tree?

Reply:
What problem would you solve by doing so?

Reply:
The following test fails because of this change by the way:

<img width="560" alt="screenshot" src="https://user-images.githubusercontent.com/594356/29714017-70c27656-89a1-11e7-9901-6642f9c23018.png">

Reply:
Well this is in the `login` branch which doesn't make sense since this is a `signup` feature too now.